RecoLocalTracker/SiPixelRecHits/src/PixelCPEFast.cc

```diff
@@ -68,13 +70,19 @@ PixelCPEFast::PixelCPEFast(edm::ParameterSet const & conf,
 
 const pixelCPEforGPU::ParamsOnGPU *PixelCPEFast::getGPUProductAsync(cuda::stream_t<>& cudaStream) const {
   const auto& data = gpuData_.dataForCurrentDeviceAsync(cudaStream, [this](GPUData& data, cuda::stream_t<>& stream) {
+
+      std::cout << "coping pixelCPEforGPU" << std::endl;
+      //here or above???
```

Comment:
If you want it to print out when the transfer is initiated, this is the correct place. ("above" outside of the lambda would print every time the product is asked from)

Reply:
yes, indeed. I wanted just to make sure it was really done once
(I was planning to fill here some __constant__ stuff, but does not work across libraries)